src/org/rascalmpl/interpreter/Evaluator.java

```diff
@@ -1146,107 +1150,82 @@ public Set<String> reloadModules(IRascalMonitor monitor, Set<String> names, ISou
     private void reloadModules(IRascalMonitor monitor, Set<String> names, ISourceLocation errorLocation, boolean recurseToExtending, Set<String> affectedModules) {
         SaveWarningsMonitor wrapped = new SaveWarningsMonitor(monitor, getErrorPrinter());
         IRascalMonitor old = setMonitor(wrapped);
+
         try {
             Set<String> onHeap = new HashSet<>();
             Set<String> extendingModules = new HashSet<>();
             PrintWriter errStream = getErrorPrinter();
-
-            try {
-                monitor.jobStart("Cleaning modules", names.size());
```

Comment:
Without that clean up, you would not be able to test if the progress bar was working correctly. That's because the previous progress report monitors were broken in different was and "bs in = bs out".